test/SlowTests/Voron/Bugs/FlushingToDataFile.cs

```diff
@@ -27,72 +27,6 @@ protected override void Configure(StorageEnvironmentOptions options)
             options.MaxLogFileSize = 2 * Constants.Storage.PageSize;
         }
 
-
-        [Fact]
-        public void SkipUpdateDatabaseStateAfterSyncIfNewerSyncCalled()
-        {
-            UpdateDatabaseStateAfterSyncIfNewerSyncCalled(false);
-        }
-
-        [Fact]
-        public void FailOnRaceConditionAtUpdateDatabaseStateAfterSyncIfNewerSyncCalled()
-        {
-            Assert.Throws<InvalidOperationException>(() => UpdateDatabaseStateAfterSyncIfNewerSyncCalled(true));
-        }
```

Comment:
Do we really need to remove them? Shouldn't we adjust to the current behavior? Maybe create new ones to verify how it currently works. I'm really reluctant to remove any  Voron tests unless there is a really good reason.

Please check the original issue: http://issues.hibernatingrhinos.com/issue/RavenDB-8515

Reply:
Yes, we want to keep them, in spirit, at least.

Reply:
I have looked at this for a long time and correct me if I'm wrong.
[These tests are check for the possibility of one sync get to `UpdateDatabaseStateAfterSync`
while another one started a new sync](https://github.com/ravendb/ravendb/pull/8148/files#diff-33af42ec0439152b32cc47c6739f3211L910).
but there is `_fsyncLock` that need to be obtained( [before](https://github.com/ravendb/ravendb/pull/8148/files#diff-33af42ec0439152b32cc47c6739f3211L980)/ [after](https://github.com/ravendb/ravendb/pull/8148/files#diff-33af42ec0439152b32cc47c6739f3211R829) ). There is [code](https://github.com/ravendb/ravendb/pull/8148/files#diff-33af42ec0439152b32cc47c6739f3211L870) in the program that dedicated just for the [test](https://github.com/ravendb/ravendb/pull/8148/files#diff-d783b9bf1a95ac3207ff8ef49e16f12fL70) and if I remove the code the test will not compile.

Reply:
We might not to have _exactly_ this tests but modify it or write something similar that will assure we won't suffer from the issue that made we added this test.

Note that `UpdateDatabaseStateAfterSync` calls `UpdateFileHeaderAfterDataFileSync(` and it's important to ensure we won't call it with outdated values (`SyncOperation._lastSyncedJournal` etc)

How do we prevent from happening it now?

@aviviadi could you take a look here?

Reply:
@haludi Please check above comment, in particular "How do we prevent from happening it now?". I think it's handled by `LockTaskResponsible` now but please confirm